dev-java/browserlauncher2: EAPI 7, min java 1.8

vaukai opened the pull request:
With `doc` enabled it happens that:
```
CVV: 8
Checked: 46 Good: 46 Bad: 0
 * Verbose logging for "java-pkg_dojar" function
 * Jar file(s) destination: /usr/share/browserlauncher2-1.0/lib
 * Jar file(s) created: /var/tmp/portage/dev-java/browserlauncher2-1.3-r2/temp/browserlauncher2.jar
 * Complete command:
 * java-pkg_dojar /var/tmp/portage/dev-java/browserlauncher2-1.3-r2/temp/browserlauncher2.jar
 * ERROR: dev-java/browserlauncher2-1.3-r2::gentoo failed (install phase):
 *   api does not exist, or isn't a directory!
```

Wow, before setting `JAVA_ANT_ENCODING="ISO-8859-1"` an error message popped up in compile phase.
Now, the error is still there but no longer the error message in compile phase.

```
The ' characters around the executable and arguments are
not part of the command.
  [javadoc] /var/tmp/portage/dev-java/browserlauncher2-1.3-r2/work/source/at/jta/Regor.java:355: error: unmappable character for encoding UTF8
  [javadoc] Loading source files for package at.jta...
  [javadoc] Loading source files for package edu.stanford.ejalbert...
  [javadoc]    * Will man den defaulteintrag �ndern, so muss man valueName "" �bergeben
  [javadoc]                                  ^
  [javadoc] /var/tmp/portage/dev-java/browserlauncher2-1.3-r2/work/source/at/jta/Regor.java:355: error: unmappable character for encoding UTF8
  [javadoc]    * Will man den defaulteintrag �ndern, so muss man valueName "" �bergeben
  [javadoc]                                                                   ^
  [javadoc] /var/tmp/portage/dev-java/browserlauncher2-1.3-r2/work/source/at/jta/Regor.java:482: error: unmappable character for encoding UTF8
  [javadoc]    * doesnt have a high access level (so maybe creating or deleting a key/value wouldn�t be successful)
  [javadoc]                                                                                       ^
  [javadoc] Loading source files for package edu.stanford.ejalbert.browserprefui...
  [javadoc] Loading source files for package edu.stanford.ejalbert.exception...
  [javadoc] Loading source files for package edu.stanford.ejalbert.exceptionhandler...
  [javadoc] Loading source files for package edu.stanford.ejalbert.launching...
  [javadoc] Loading source files for package edu.stanford.ejalbert.launching.macos...
  [javadoc] Loading source files for package edu.stanford.ejalbert.launching.misc...
  [javadoc] Loading source files for package edu.stanford.ejalbert.launching.utils...
  [javadoc] Loading source files for package edu.stanford.ejalbert.launching.windows...
  [javadoc] Loading source files for package edu.stanford.ejalbert.testing...
  [javadoc] Loading source files for package net.sf.wraplog...
  [javadoc] 3 errors
  [javadoc] No javadoc created, no need to post-process anything

BUILD SUCCESSFUL
Total time: 1 second
>>> Source compiled.
```
Ehm
`3 errors`, `BUILD SUCCESSFUL`

Should I try converting that source file [as done here for some docs](https://github.com/gentoo/gentoo/blob/master/sys-block/scsiadd/scsiadd-1.97-r2.ebuild#L29-L31)?

fordfrog commented:
i will have to have a look at it. anyway, with wrong encoding, javadoc is not created in this case, which should fail, but it does not... :-)

fordfrog commented:
or you might wanna try switching it to `java-pkg-simple` to avoid `ant`. this package is really old... :-|

vaukai commented:
> i will have to have a look at it. anyway, with wrong encoding, javadoc is not created in this case, which should fail, but it does not... :-)

Yes, very strange. Confusing and misleading.  But I just solved it, finally!

fordfrog commented:
@vaukai do you use some kind of chat app, like signal or telegram or whatsapp or something like that? i think it might come handy to chat somehow directly. wrt the fix, great, will have a look at it, proly tomorrow :-)

vaukai commented:
> @vaukai do you use some kind of chat app, like signal 
Yep, pls find mobile no on impress of `pogatzki.net`

fordfrog commented on dev-java/browserlauncher2/browserlauncher2-1.3-r2.ebuild:
```diff
@@ -0,0 +1,49 @@
+	iconv -f ISO-8859-1 -t UTF8 -o /var/tmp/portage/dev-java/browserlauncher2-1.3-r2/work/source/at/jta/Regor.java~ \
+		/var/tmp/portage/dev-java/browserlauncher2-1.3-r2/work/source/at/jta/Regor.java || die "recoding failed"
+	mv -f /var/tmp/portage/dev-java/browserlauncher2-1.3-r2/work/source/at/jta/Regor.java{~,} || die "cannot rename"
+}
```
cool! just should use `${S}` instead of hardcoded paths :-)

vaukai replied:
Of course. Could you  change it on merge, please?

fordfrog commented:
@vaukai thanks! i fixed the hardcoded paths and switched to stable (cc83ee1b103be24d06b01bb6dd9a0affd7e0bfdf), and dropped the old.

vaukai commented:
> @vaukai thanks! i fixed the hardcoded paths and switched to stable ([cc83ee1](https://github.com/gentoo/gentoo/commit/cc83ee1b103be24d06b01bb6dd9a0affd7e0bfdf)), and dropped the old.

One less ...
```
git --no-pager grep -Elo "virtual/(jre|jdk)-1.[2-5]" | sort | grep '^dev-java' | wc -l
109
```

fordfrog commented:
can't believe my eyes, it was 195 when i sent the email to gentoo-dev. it's close to half of the number. and some of those are already resolved, just waiting on stabilization and cleanup.
